lxd/patches.go

```diff
@@ -65,3 +71,21 @@ func patchesApplyAll(d *Daemon) error {
 }
 
 // Patches begin here
+func patchInvalidProfileNames(name string, d *Daemon) error {
+	profiles, err := dbProfiles(d.db)
+	if err != nil {
+		return err
+	}
+
+	for _, profile := range profiles {
+		if strings.Contains(profile, "/") || shared.StringInSlice(profile, []string{".", ".."}) {
+			shared.Log.Info("Removing unreachable profile (invalid name)", log.Ctx{"name": profile})
+			err := dbProfileDelete(d.db, profile)
```

Comment:
Nope, it'll un-apply it from the container. That's why I added the log message so that there's a trace that we've done so.

A profile in that state cannot be dumped, modified or deleted so even if someone managed to create one and apply it to containers, it'd most likely be an empty profile so that shouldn't actually harm anyone.